chromadb/server/fastapi/__init__.py

```diff
@@ -451,7 +453,7 @@ def process_create_tenant(request: Request, raw_body: bytes) -> None:
         await to_thread.run_sync(
             process_create_tenant,
             request,
-            await request.body(),
+            body,
```

Comment:
This breaks the async model. FastAPI blocks the main thread while serializing bodies, which impacts performance in a big way. I think we can expose types in the `__init__`.  Let me check.

Reply:
<img width="1525" alt="image" src="https://github.com/chroma-core/chroma/assets/1157440/1bfdee3e-5f68-42f2-a82a-20bf6272c782">


with the following:

```python
self.router.add_api_route(
            "/api/v1/collections/{collection_id}/add",
            self.add,
            methods=["POST"],
            status_code=status.HTTP_201_CREATED,
            response_model=bool,
        )

...
 @trace_method("FastAPI.add", OpenTelemetryGranularity.OPERATION)
    async def add(self, request: Request, collection_id: str, add_request: AddEmbedding = Body(...)) -> bool:
        try:

            def process_add(request: Request, raw_body: bytes) -> bool:
                add = AddEmbedding.model_validate(orjson.loads(raw_body))
                self.auth_and_get_tenant_and_database_for_request(
                    request.headers,
                    AuthzAction.ADD,
                    None,
                    None,
                    collection_id,
                )
```

Reply:
@tazarov oh interesting. this is good to know! please let m know